range.js

```diff
@@ -44,63 +44,129 @@ $.fn.range = function(parameters) {
 				
 				inner,
 				thumb,
+				rthumb,
+				thumbpos,
+				rthumbpos,
+				thumbval,
+				rthumbval,
 				trackLeft,
 				precision,
-				
 				module
 			;
-			
 			module = {
 				
 				initialize: function() {
-					module.instantiate();
 					module.sanitize();
+					module.instantiate();
+					$(window).resize(function() {
+						module.instantiate();
+					});
 				},
 				
 				instantiate: function() {
 					instance = module;
 					$module
 						.data(moduleNamespace, module)
 					;
-					$(element).html("<div class='inner'><div class='track'></div><div class='track-fill'></div><div class='thumb'></div></div>");
-					inner = $(element).children('.inner')[0];
-					thumb = $(element).find('.thumb')[0];
-					trackLeft = $(element).find('.track-fill')[0];
-					// find precision of step, used in calculating the value
-					module.determinePrecision();
-					// set start location
-					module.setValuePosition(settings.start);
-					// event listeners
-					$(element).find('.track, .thumb, .inner').on('mousedown', function(event) {
-						event.stopImmediatePropagation();
-						event.preventDefault();
-						$(this).closest(".range").trigger('mousedown', event);
-					});
-					$(element).find('.track, .thumb, .inner').on('touchstart', function(event) {
-						event.stopImmediatePropagation();
-						event.preventDefault();
-						$(this).closest(".range").trigger('touchstart', event);
-					});
-					$(element).on('mousedown', function(event, originalEvent) {
-						module.rangeMousedown(event, false, originalEvent);
-					});
-					$(element).on('touchstart', function(event, originalEvent) {
-						module.rangeMousedown(event, true, originalEvent);
-					});
+					if (settings.values) {
```

Comment:
Most of the code inside this `if` block is a near duplicate of the code in the following `else`. I think we could dry it up by wrapping the reused portions inside methods.